File: HACKATHON_PRACTICE - BACKEND/BACKEND/BACKEND/src/main/java/BACKEND/BACKEND/Authh/Auth/backend/modules/auth/controller/authController.js
const authService = require('../service/authService');

const registerUser = async (req, res) => {
    try {
        const { name, email, password } = req.body;
        const user = await authService.registerUser({ name, email, password });

        res.status(201).json({
            success: true,
            data: user,
            message: 'User registered successfully'
        });
    } catch (error) {
        res.status(400).json({
            success: false,
            data: null,
            message: error.message
        });
    }
};

const loginUser = async (req, res) => {
    try {
        const { email, password } = req.body;
        const { token, userId } = await authService.loginUser(email, password);

        res.status(200).json({
            success: true,
            data: {
                token,
                userId
            },
            message: 'Login successful'
        });
    } catch (error) {
        res.status(401).json({
            success: false,
            data: null,
            message: error.message
        });
    }
};

module.exports = {
    registerUser,
    loginUser
};

File: HACKATHON_PRACTICE - BACKEND/BACKEND/BACKEND/src/main/java/BACKEND/BACKEND/auth/controller/AuthController.java
package BACKEND.BACKEND.auth.controller;

import BACKEND.BACKEND.auth.dto.ApiResponse;
import BACKEND.BACKEND.auth.dto.AuthResponse;
import BACKEND.BACKEND.auth.dto.LoginRequest;
import BACKEND.BACKEND.auth.dto.RegisterRequest;
import BACKEND.BACKEND.auth.service.AuthService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;

    @PostMapping("/register")
    public ResponseEntity<ApiResponse<AuthResponse>> register(@RequestBody RegisterRequest request) {
        try {
            AuthResponse response = authService.register(request);
            return ResponseEntity.ok(ApiResponse.success(response, "User registered successfully"));
        } catch (Exception e) {
            return ResponseEntity.badRequest().body(ApiResponse.error(e.getMessage()));
        }
    }

    @PostMapping("/login")
    public ResponseEntity<ApiResponse<AuthResponse>> login(@RequestBody LoginRequest request) {
        try {
            AuthResponse response = authService.login(request);
            return ResponseEntity.ok(ApiResponse.success(response, "User logged in successfully"));
        } catch (Exception e) {
            return ResponseEntity.status(401).body(ApiResponse.error(e.getMessage()));
        }
    }
}
